pandera/schemas.py

```diff
@@ -707,13 +859,25 @@ def rename_columns(self, rename_dict: Dict[str, str]) -> "DataFrameSchema":
                 f"Keys {not_in_cols} not found in schema columns!"
             )
 
+        # ensure all new keys are not present in the current column names
+        already_in_columns: List[str] = [
+            x for x in rename_dict.values() if x in new_schema.columns.keys()
+        ]
+        if already_in_columns:
+            raise errors.SchemaInitError(
+                f"Keys {already_in_columns} already found in schema columns!"
+            )
+
         # We iterate over the existing columns dict and replace those keys
         # that exist in the rename_dict
+
         new_columns = {
-            (
-                rename_dict[col_name] if col_name in rename_dict else col_name
-            ): col_attrs
-            for col_name, col_attrs in self.columns.items()
+            (rename_dict[col_name] if col_name in rename_dict else col_name): (
```

Comment:
The previous implementation left the name unchanged. So if you changed a name from col1 to col2, the name parameter remained col1. I understand why changing names is undesirable, but this is surely not intended?

Reply:
ah, nice! thanks for catching this